src/endpoints/groups.js

```diff
@@ -18,16 +19,8 @@ const setGroup = (nameExtractor) => (req, res, next) => {
  */
 
 const optionsGroupSettings = (req, res) => {
-  authz.assertAuthorized(req.user, authz.actions.Read, req.context.group);
-
-  const allowedMethods = ["OPTIONS", "GET", "HEAD"];
-
-  if (authz.authorized(req.user, authz.actions.Write, req.context.group)) {
-    allowedMethods.push("PUT", "DELETE");
-  }
-
-  res.set("Allow", allowedMethods);
-  return res.status(204).end();
+  req.context.authzObject = req.context.group;
+  return optionsAuthzObject(req, res);
```

Comment:
I find this to be a weird construction. It's an indirect passing of an argument to `optionsAuthzObject`… but the call to the function is right there, so it could just be passed directly.

I think there's a way to do this without the weirdness by using a single general function for setting `req.context.authzObject` (e.g. similar to how `setSource()` works) and then using `optionsAuthzObject()` in handler chains (e.g. similar to how `ifDatasetExists()` works). Or similar approaches with slightly different interfaces. In any case, `optionsGroupSettings()` would go away in favor of the general thing.

Might be easiest for me to describe by implementing it…

Maybe let's drop 2bbed5d7a27fd5a858e3c98393f6c1fdb2e2e651 for now, merge this, and then we can circle back to this later?

Reply:
> Might be easiest for me to describe by implementing it…

Yeah, I'm a little lost 😅 

> Maybe let's drop https://github.com/nextstrain/nextstrain.org/commit/2bbed5d7a27fd5a858e3c98393f6c1fdb2e2e651 for now, merge this, and then we can circle back to this later?

Sounds good to me.

Reply:
Dropped 2bbed5d7a27fd5a858e3c98393f6c1fdb2e2e651 in latest push and merging.

Reply:
> Yeah, I'm a little lost :sweat_smile:

:upside_down_face: Yeah… sorry! I realized I wasn't being super clear and that it'd be clearest to show rather than say.

> a single general function for setting `req.context.authzObject` _[…]_ Or similar approaches with slightly different interfaces.

I went with the "similar approach with slightly different interface" option (har har): https://github.com/nextstrain/nextstrain.org/pull/686